shotover-proxy/src/transforms/protect/mod.rs

```diff
@@ -30,6 +33,161 @@ pub struct Protect {
     key_id: String,
 }
 
+impl Protect {
+    /// encodes a Protected object into a byte array.  This is here to centeralize the serde for
+    /// the Protected object.
+    /// Returns an error if a Plaintext Protected object is passed
+    fn encode(protected: &Protected) -> Result<Vec<u8>> {
+        match protected {
+            Protected::Plaintext(_) => Err(anyhow!("can not encode plain text")),
+            Protected::Ciphertext { .. } => Ok(serde_json::to_vec(protected)?),
+        }
+    }
+
+    /// decodes a byte array into the Protected object.  This is here to centeralize the serde for
+    /// the Protected object.
+    fn decode(data: &[u8]) -> Result<Protected> {
+        Ok(serde_json::from_slice(data)?)
```

Comment:
I am pretty sure the serde encoding was a hack to let us store binary data in a string type due to limitations in the old cql parser, we should be able to get rid of it and just directly use the bincode encoded data.